sdk/servicebus/service-bus/src/receiver.ts

```diff
@@ -554,9 +569,11 @@ export class SessionReceiver {
    * @returns {Promise<void>}
    */
   async close(): Promise<void> {
+    this._isClosed = true;
```

Comment:
What are your thoughts on setting `_isClosed` to `true` right away like this vs after all the receivers are closed like the way it is done in the `close()` call of the `Receiver` class?

Reply:
This flag is specific to the SessionReceiver class and don't think it would make much of a difference.
To address the race condition, we likely don't want to miss updating this in case `_messageSession.close()` fails and throws an error.

Reply:
Yes, and there is a similar flag in the Receiver class, and thus my question comparing the way we set the 2 in the close() method of the 2 classes. I want to follow the same model for both classes regardless of whether we choose to set it this way or in a `finally` block